packages/frontend-2/pages/settings/workspaces/[slug]/security.vue

```diff
@@ -258,23 +260,24 @@ const isDomainProtectionEnabled = computed({
 })
 
 const isDomainDiscoverabilityEnabled = computed({
-  get: () => result.value?.workspace.discoverabilityEnabled || false,
+  get: () => workspace.value?.discoverabilityEnabled || false,
   set: async (newVal) => {
+    if (!workspace.value?.id) return
     const mutationResult = await apollo.mutate({
       mutation: SettingsUpdateWorkspaceSecurityDocument,
       variables: {
         input: {
-          id: props.workspaceId,
+          id: workspace.value?.id,
           discoverabilityEnabled: newVal
         }
       },
       optimisticResponse: {
```

Comment:
can we at least add a FIXME here to clean this up? the optimisticResponse still doesn't make any sense to me and cache.modify should be replaced with modifyObjectField

Reply:
Will do 👍

Reply:
@fabis94 I changed this to use normal mutations, manual updating the cache and optismisticResponse is now not needed anymore